addons/account/models/account_move_send.py

```diff
@@ -733,6 +738,30 @@ def _check_sending_data(self, moves, **custom_settings):
             for sending_method in custom_settings.get('sending_methods', [])
         ) if 'sending_methods' in custom_settings else True
 
+    # As we are often sending invoices to the government, we want to make sure that
+    # the record is correctly stored in the db before sending it
+    def after_commit(func):
+        @wraps(func)
+        def wrapped(self, *args, **kwargs):
+            if self.env['account.move']._can_commit():
+                @self.env.cr.postcommit.add
+                def called_after():
+                    try:
+                        func(self, *args, **kwargs)
```

Comment:
We somehow need to commit at this point, postcommit hooks don't commit by itselves.

Also the sending of the invoice email with be deferred at postcommit (will run next) but this is done is a separate commit (see [`send_emails_with_new_cursor()`](<https://github.com/odoo/odoo/blob/saas-18.3/addons/mail/models/mail_mail.py#L582>). So if we don't commit, the attachments we generated won't be stored nor the mail.mail we just created (won't be seen within `send_emails_with_new_cursor()` transaction an no email will be send).
```suggestion
                        func(self, *args, **kwargs)
                        self.env.cr.commit()
```